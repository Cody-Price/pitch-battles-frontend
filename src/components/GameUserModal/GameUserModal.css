.game-user-modal {
  width: 180px;
  height: 240px;
  position: absolute;
  left: -300px;
  top: 110px;
  background: url("../Game/images/big-modal-frame.png");
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  transition: 0.3s;
  font-family: "chronotype";
}

.game-user-modal.true {
  left: 0px;
}

.game-user-modal button {
  padding-top: 3px;
  position: absolute;
  font-family: "chronotype";
  display: block;
  font-size: 1.3rem;
  margin: auto;
  background: #f2f2f2;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
  border: 1px solid #333;
  text-align: center;
  width: 80px;
}

.game-user-modal button:focus {
  outline: none;
}

.game-user-modal button:active {
  box-shadow: none;
}

.game-user-modal button:hover {
  cursor: pointer;
}

.exit-game-button {
  bottom: 30px;
  width: 50px;
  left: 0;
  right: 0;
}

.reset-game-button {
  bottom: 60px;
  width: 60px;
  left: 0;
  right: 0;
}

.game-modal-username {
  font-size: 1.5rem;
  margin: 2rem 0 0;
}

.current-instrument-label {
  font-size: 1.3rem;
  font-weight: 700;
  margin: 1rem 0 0;
}

.current-instrument {
  font-size: 1.2rem;
  margin: 0;
}

.exiting-message,
.resetting-message {
  position: absolute;
  /* left: -400px; */
  top: 85px;
  right: 0;
  left: -300px;
  margin: auto;
  transition: 0.1s;
}

.resetting-message {
  top: 101px;
}

.exiting-message.true,
.resetting-message.true {
  left: 0;
}
