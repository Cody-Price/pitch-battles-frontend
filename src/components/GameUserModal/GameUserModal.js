import React, { Component } from "react";
import PropTypes from "prop-types";

import "./GameUserModal.css";

class GameUserModal extends Component {
  render() {
    return (
      <aside className={`game-user-modal ${this.props.status}`}>
        <h4 className="game-modal-username">{this.props.username}</h4>
        <p className="current-instrument-label">instrument</p>
        <p className="current-instrument">{this.props.instrument}</p>
        <p className={`${this.props.resetting} resetting-message`}>
          resetting...
        </p>
        <p className={`${this.props.exiting} exiting-message`}>exiting...</p>
        <button onClick={this.props.reset} className="reset-game-button">
          reset
        </button>
        <button
          onClick={() => this.props.endGame(false)}
          className="exit-game-button"
        >
          exit
        </button>
      </aside>
    );
  }
}

export default GameUserModal;

GameUserModal.propTypes = {
  status: PropTypes.bool,
  username: PropTypes.string,
  instrument: PropTypes.string,
  resetting: PropTypes.bool,
  exiting: PropTypes.bool,
  reset: PropTypes.func,
  endGame: PropTypes.func
};
