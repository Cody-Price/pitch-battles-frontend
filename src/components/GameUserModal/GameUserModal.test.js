import React from "react";
import ReactDOM from "react-dom";
import GameUserModal from "./GameUserModal";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("GameUserModal", () => {
  let wrapper;
  const mockStatus = true;
  const mockUsername = "Joey";
  const mockInstrument = "triangle";
  let mockReset;
  let mockEndGame;

  beforeEach(() => {
    mockReset = jest.fn();
    mockEndGame = jest.fn();

    wrapper = shallow(
      <GameUserModal
        status={mockStatus}
        username={mockUsername}
        instrument={mockInstrument}
        reset={mockReset}
        endGame={mockEndGame}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call reset on click", () => {
    wrapper.find(".reset-game-button").simulate("click");

    expect(mockReset).toHaveBeenCalled();
  });

  it("should call endGame on click", () => {
    wrapper.find(".exit-game-button").simulate("click");

    expect(mockEndGame).toHaveBeenCalledWith(false);
  });
});
