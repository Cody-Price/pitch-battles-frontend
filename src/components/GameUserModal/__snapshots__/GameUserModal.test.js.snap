// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameUserModal should match the snapshot 1`] = `
<aside
  className="game-user-modal true"
>
  <h4
    className="game-modal-username"
  >
    Joey
  </h4>
  <p
    className="current-instrument-label"
  >
    instrument
  </p>
  <p
    className="current-instrument"
  >
    triangle
  </p>
  <p
    className="undefined resetting-message"
  >
    resetting...
  </p>
  <p
    className="undefined exiting-message"
  >
    exiting...
  </p>
  <button
    className="reset-game-button"
    onClick={[MockFunction]}
  >
    reset
  </button>
  <button
    className="exit-game-button"
    onClick={[Function]}
  >
    exit
  </button>
</aside>
`;
